sound/soc/intel/boards/Kconfig

```diff
@@ -270,6 +270,17 @@ config SND_SOC_INTEL_BXT_TDF8532_MACH
 	  Say Y if you have such a device.
 	  If unsure select "N".
 
+config SND_SOC_INTEL_SKL_HDA_DSP_GENERIC_MACH
+	tristate "SKL/KBL/BXT/APL with HDA Codecs"
+	select SND_SOC_HDAC_HDMI
+	select SND_SOC_HDAC_HDA
+	select SND_SOC_SOF_HDA if SND_SOC_SOF_HDA_COMMON
```

Comment:
this looks odd. why is this needed in the boards Kconfig?
Looks further, this makes no sense, in the same patch you have 

select SND_SOC_SOF_HDA if SND_SOC_SOF_HDA_COMMON


 +config SND_SOC_SOF_HDA
 +	tristate
 +	depends on SND_SOC_SOF_HDA_COMMON

so you've expressed a dependency twice for no good reason

Reply:
um, this do need rewrite, I will do that. But before that, can you explain to me that is selecting 2 machine driver(e.g. BXT_TDF8532_MACH and SKL_HDA_DSP_GENERIC_MACH) together is our intention?